json-schema/json-schema-circe/src/main/scala/endpoints/circe/JsonSchemas.scala

```diff
@@ -53,6 +53,8 @@ trait  JsonSchemas
       }
   }
 
+  def named[A, S[T] <: JsonSchema[T]](schema: S[A], name: String): S[A] = schema
```

Comment:
What happens if we write 'named(named(s, "foo"), "bar")'? I think it would be nice to have a type signature preventing this kind of constructions.

Reply:
It depends on the interpreter which have possibility to examine existing name (if any). Currently `openapi` interpreter just overwrites the name.
I agree it would be nice to prevent it, but solutions coming to my mind make the signature too complex that I'm not sure the benefit is worth the costs.
Maybe you have some ideas?

Reply:
The solution I see would be to introduce to new type members:

~~~ scala
type UnnamedSchema[A] <: JsonSchema[A]
type NamedSchema[A, S[T] <: UnnamedSchema[T]] <: JsonSchema[A]
// And then:
def named[A, S[T] <: UnnamedSchema[T]](schema: S[A], name: String): NamedSchema[A, S]
// Also, existing Tagged and Record types are refined:
type Tagged[A] <: UnnamedSchema[A]
type Record[A] <: UnnamedSchema[A]
~~~

But I think it’s better to live with your solution for now. We could implement this more sophisticated one later if we feel like we really need it! Is that ok for you?